internal/pkg/source/source.go

```diff
@@ -109,13 +109,28 @@ func (a *Source) visitOperands(operands []*ssa.Value) {
 		if !ok || a.marked[n] {
 			continue
 		}
-		al, ok := (*o).(*ssa.Alloc)
-		if !ok || al.Comment == "slicelit" || a.config.IsSource(al.Type()) {
-			a.dfs(n)
+		// An Alloc represents the allocation of space for a variable. If a Node is an Alloc,
+		// and the thing being allocated is not an array, then either:
+		// a) it is a Source value, in which case it will get its own DFS when sourcesFromBlocks
+		//    finds this Alloc
+		// b) it is not a Source value, in which case we should not DFS through it.
+		// However, if the Alloc is an array, then that means the source that we are DFSing from
+		// is being placed into an array, slice or varags, so we do need to keep DFSing.
+		al, isAlloc := (*o).(*ssa.Alloc)
+		if isAlloc {
+			if _, isArray := utils.Dereference(al.Type()).(*types.Array); !isArray {
+				return
+			}
 		}
```

Comment:
I don't know why I didn't notice it when the line was above the comment block and I suggested you move it down...

You can in-line the declaration on both of these checks.
```golang
if al, isAlloc := (*o).(*ssa.Alloc); isAlloc {
	if _, isArray := utils.Dereference(al.Type()).(*types.Array); !isArray {
		return
	}
}
```

And actually, since `Type()` is on the interface `o` satisfies, you could even go with 

```golang
if _, isAlloc := (*o).(*ssa.Alloc); isAlloc {
	if _, isArray := utils.Dereference((*o).Type()).(*types.Array); !isArray {
		return
	}
}
```

Although I think the first block reads better.

Reply:
Good stuff, thanks! Done.

I actually don't really like this style of `if` statement. It's good because it limits the scope of the variables, but I find it hard to read. Anyhoo, this seems to be universally accepted as idiomatic so I'll just get used to it.

Thanks for including the second block, very informative.